### PR TITLE
EVG-18014: Refactor EditSpawnHost modal to use SpruceForm

### PR DESCRIPTION
**Pull request by minnakt**
EVG-18014

### Description
This PR refactors the code around the `EditSpawnHost` modal so that it uses `SpruceForm`. This allows us to delete a good amount of files around this part of the codebase.

This PR also addresses [EVG-18003](https://jira.mongodb.org/browse/EVG-18003).

### Screenshots
I changed the design of the modal slightly to work nicely with the form. These changes make the modal adhere better to design systems [documentation](https://www.mongodb.design/component/text-input/guidelines/), which says that we shouldn't be hiding information in tooltips for user inputs.

Screen Shot 2022-12-01 at 9 08 52 AM

### Testing
- Added `utils.test.ts`
- Added `transformers.test.ts`
- Manually testing on staging to see if my changes went through

### CODE REVIEW

File: src/components/Spawn/editHostModal/FieldTemplates/ExpirationRow.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import styled from "@emotion/styled";
+import { ObjectFieldTemplateProps } from "@rjsf/core";
+import { size } from "constants/tokens";
+import { form } from "pages/projectSettings/tabs/utils";
+
+const { getFields } = form;
```
Comment (khelif96) [inline src/components/Spawn/editHostModal/FieldTemplates/ExpirationRow.tsx:6]: So sorry I didn't know this in past code reviews. This should be moved to a more global directory. Possibly In a `spruceform/utils` directory or something along those lines.

File: src/components/Spawn/editHostModal/getFormSchema.tsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
+import { css } from "@emotion/react";
+import { add } from "date-fns";
+import widgets from "components/SpruceForm/Widgets";
+import { AntdSelect } from "components/SpruceForm/Widgets/AntdWidgets";
+import { LeafyGreenTextArea } from "components/SpruceForm/Widgets/LeafyGreenWidgets";
+import { InputLabel, StyledLink } from "components/styles";
+import { windowsPasswordRulesURL } from "constants/externalResources";
+import { GetMyPublicKeysQuery, MyVolumesQuery } from "gql/generated/types";
+import { GetFormSchema } from "pages/projectSettings/tabs/types";
```
Comment (khelif96) [inline src/components/Spawn/editHostModal/getFormSchema.tsx:9]: Same comment about this we shouldn't be importing this from the `pages/projectSettings` folder.

File: src/components/Spawn/editHostModal/useLoadFormData.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  GET_INSTANCE_TYPES,
+  GET_MY_PUBLIC_KEYS,
+  GET_MY_VOLUMES,
+} from "gql/queries";
+import { useDisableSpawnExpirationCheckbox, useSpruceConfig } from "hooks";
+import { MyHost } from "types/spawn";
+import { getNoExpirationCheckboxTooltipCopy } from "../utils";
+
+export const useLoadFormData = (host: MyHost) => {
```
Comment (khelif96) [inline src/components/Spawn/editHostModal/useLoadFormData.ts:19]: Great idea with the hook!

**Review by khelif96 — APPROVED** [review]
All around great refactor! Just had a comment about folder structure for an import.

File: src/components/Spawn/editHostModal/useLoadFormData.ts (hunk 3 above)
Comment (minnakt) [reply src/components/Spawn/editHostModal/useLoadFormData.ts:19]: all credit goes to arjun who did it before me :P

**Pull request merged**